subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -326,13 +336,23 @@ uint8_t ll_df_set_cl_cte_tx_enable(uint8_t adv_handle, uint8_t cte_enable)
 		cte_info.time = df_cfg->cte_length;
 		pdu_data.field_data = (uint8_t *)&cte_info;
 		pdu_data.extra_data = df_cfg;
-		err = ull_adv_sync_pdu_set_clear(adv,
+
+		err = ull_adv_sync_pdu_alloc(adv, 0,
+					     ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+					     NULL, &pdu_prev, &pdu, &ter_idx);
+		if (err) {
+			return err;
+		}
+
+		err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
 						 ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
-						 0, &pdu_data, &ter_idx);
+						 0, &pdu_data);
 		if (err) {
 			return err;
 		}
 
+		lll_adv_sync_data_enqueue(lll_sync, ter_idx);
```

Comment:
`lll_adv_sync_data_enqueue` is called once, just after end of if-else statement. It covers both parts of condition. This call is not needed here.